libraries/GCS_MAVLink/GCS.cpp

```diff
@@ -19,13 +19,108 @@
 #include <AP_OpticalFlow/AP_OpticalFlow.h>
 #include <AP_GPS/AP_GPS.h>
 #include <RC_Channel/RC_Channel.h>
+#include <AP_Vehicle/AP_Vehicle_Type.h>
 
 #include "MissionItemProtocol_Waypoints.h"
 #include "MissionItemProtocol_Rally.h"
 #include "MissionItemProtocol_Fence.h"
 
 extern const AP_HAL::HAL& hal;
 
+#ifndef MAV_SYSID_DEFAULT
+#if APM_BUILD_TYPE(APM_BUILD_AntennaTracker)
+#define MAV_SYSID_DEFAULT 2
+#else
+#define MAV_SYSID_DEFAULT 1
+#endif  // APM_BUILD_TYPE(APM_BUILD_AntennaTracker)
+#endif  // defined(MAV_SYSID_DEFAULT)
+
+const AP_Param::GroupInfo GCS::var_info[] {
+    // @Param: _SYSID
+    // @DisplayName: MAVLink system ID of this vehicle
+    // @Description: Allows setting an individual MAVLink system id for this vehicle to distinguish it from others on the same network
+    // @Range: 1 255
+    // @User: Advanced
+    AP_GROUPINFO("_SYSID",    1,     GCS,  sysid,  MAV_SYSID_DEFAULT),
+
+    // @Param: _MYGCS_SYSID
+    // @DisplayName: My ground station number
+    // @Description: This controls whether packets from other than the expected GCS system ID will be accepted
+    // @Range: 1 255
+    // @Increment: 1
+    // @User: Advanced
+    AP_GROUPINFO("_MYGCS_SYSID",    2,      GCS, mav_mygcs_sysid, 255),
+
+    // @Param: _MYGCS_ENFRCE
+    // @DisplayName: GCS sysid enforcement
+    // @Description: Allows restricting radio overrides to only come from my ground station
+    // @Values: 0:NotEnforced,1:Enforced
+    // @User: Advanced
+    AP_GROUPINFO("_MYGCS_ENFRCE",    3,      GCS, mav_mygcs_enforce, 0),
```

Comment:
discuss name with @rmackay9

Reply:
decided on MAV_OPTIONS bitmask AP_Int32 for enforce, should be easy conversion

Reply:
Changed.  Only AP_Int16 - already overkill as we only have one at the moment and haven't needed anything else what might have gone in here.